panther_hardware_interfaces/include/panther_hardware_interfaces/roboteq_data_converters.hpp

```diff
@@ -0,0 +1,282 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_HARDWARE_INTERFACES_ROBOTEQ_DATA_CONVERTERS_HPP_
+#define PANTHER_HARDWARE_INTERFACES_ROBOTEQ_DATA_CONVERTERS_HPP_
+
+#include <bitset>
+#include <cstdint>
+#include <vector>
+
+#include <panther_msgs/msg/fault_flag.hpp>
+#include <panther_msgs/msg/runtime_error.hpp>
+#include <panther_msgs/msg/script_flag.hpp>
+
+#include <panther_hardware_interfaces/roboteq_driver.hpp>
+#include <panther_hardware_interfaces/utils.hpp>
+
+namespace panther_hardware_interfaces
+{
+struct DrivetrainSettings
+{
+  float motor_torque_constant;
+  float gear_ratio;
+  float gearbox_efficiency;
+  float encoder_resolution;
+  float max_rpm_motor_speed;
+};
+
+/**
+ * @brief Class for converting velocity commands (between SI units and raw Roboteq cmd)
+ */
+class RoboteqVeloctiyCommandConverter
+{
+public:
+  RoboteqVeloctiyCommandConverter(const DrivetrainSettings & drivetrain_settings);
+  std::int32_t Convert(const float cmd) const
+  {
+    return LimitCmd(cmd * radians_per_second_to_roboteq_cmd_);
+  }
+
+private:
+  inline std::int32_t LimitCmd(const std::int32_t cmd) const
+  {
+    return std::clamp(cmd, -kMaxRoboteqCmdValue, kMaxRoboteqCmdValue);
+  }
+
+  float radians_per_second_to_roboteq_cmd_;
+  static constexpr std::int32_t kMaxRoboteqCmdValue = 1000;
+};
+
+/**
+ * @brief Class for storing and converting (between raw Roboteq data and SI units) motor
+ * state (position, velocity, torque)
+ */
+class MotorState
+{
+public:
+  MotorState(const DrivetrainSettings & drivetrain_settings);
+
+  void SetData(const RoboteqMotorState & fb) { last_state_ = fb; };
+
+  float GetPosition() const { return last_state_.pos * roboteq_pos_feedback_to_radians_; }
+  float GetVelocity() const
+  {
+    return last_state_.vel * roboteq_vel_feedback_to_radians_per_second_;
+  }
+  float GetTorque() const
+  {
+    return last_state_.current * roboteq_current_feedback_to_newton_meters_;
+  }
+
+private:
+  float roboteq_pos_feedback_to_radians_;
+  float roboteq_vel_feedback_to_radians_per_second_;
+  float roboteq_current_feedback_to_newton_meters_;
+
+  RoboteqMotorState last_state_ = {0, 0, 0};
```

Comment:
```suggestion
  RoboteqMotorState motor_state_ = {0, 0, 0};
```
Don't see point in calling it a "last" when it is the only value.